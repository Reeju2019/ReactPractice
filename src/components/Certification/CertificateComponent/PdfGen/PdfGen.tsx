import React, { useRef } from "react";
import ReactToPdf from "react-to-pdf";
import { Button } from "react-bootstrap";

/**
 * @description Props of PdfGen Component
 * @memberof PdfGen
 * @property {string} downloadButtonName Set the download button name
 * @property {string} filename Set the filename
 * @property {any} downloadRef Download ref
 * @property {any} options Options
 */
/* eslint-disable */
interface IProps {
  downloadButtonName: string;
  filename: string;
  downloadRef: any;
  options: any;
}

/**
 * @description States of PdfGen Component
 * @memberof PdfGen
 */
/* eslint-disable */
interface IState {}

/**
 * @component
 * This component was used in Certificate component
 * @description PdfGen Component
 */
class PdfGen extends React.Component<IProps, IState> {
  state: IState = {};

  /**
   * @constructor
   * @description Constructor
   * @param {IProps} props Props
   * @returns {void}
   */
  constructor(props: IProps) {
    super(props);
  }

  render() {
    console.log(this.props.downloadRef);
    console.log(this.props.filename);

    return (
      <>
        <ReactToPdf
          targetRef={this.props.downloadRef}
          filename={this.props.filename}
          options={this.props.options}
          scale={0.9}
        >
          {({ toPdf }) => (
            <Button
              variant="primary"
              className="btn-lg download-btn"
              onClick={toPdf}
            >
              {this.props.downloadButtonName}
            </Button>
          )}
        </ReactToPdf>
      </>
    );
  }
}

export { PdfGen };
export default PdfGen;
