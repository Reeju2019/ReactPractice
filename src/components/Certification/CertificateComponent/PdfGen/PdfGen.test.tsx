import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import PdfGen from "./PdfGen"

/**
 * Unit testing for PdfGen Component
 */
describe("PdfGen Component Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <PdfGen
        downloadButtonName={"download"}
        filename={"test.download.pdf"}
        downloadRef={null}
        options={{
          orientation: "p",
          unit: "mm",
          format: "a4",
          putOnlyUsedFonts: true,
        }}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <PdfGen
        downloadButtonName={"download"}
        filename={"test.download.pdf"}
        downloadRef={null}
        options={{
          orientation: "p",
          unit: "mm",
          format: "a4",
          putOnlyUsedFonts: true,
        }}
      />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
