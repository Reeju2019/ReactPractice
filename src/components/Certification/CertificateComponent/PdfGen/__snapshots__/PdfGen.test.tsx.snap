// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfGen Component Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <ReactToPdf
      filename="test.download.pdf"
      options={
        Object {
          "format": "a4",
          "orientation": "p",
          "putOnlyUsedFonts": true,
          "unit": "mm",
        }
      }
      scale={0.9}
      targetRef={null}
      x={0}
      y={0}
    >
      [Function]
    </ReactToPdf>
  </React.Fragment>,
]
`;
